### PR TITLE
cythonize the chain polynomials of posets

### PR DESCRIPTION
**Pull request by fchapoton**
for the sake of speed

### :memo: Checklist

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [x] I have created tests covering the changes.
- [x] I have updated the documentation and checked the documentation preview.

### CODE REVIEW

File: src/sage/combinat/posets/hasse_cython_flint.pyx [hunk 1]
```diff
@@ -20,6 +20,53 @@ from sage.libs.flint.fmpz_mat cimport *
 from sage.matrix.matrix_integer_dense cimport Matrix_integer_dense
 from sage.matrix.matrix_space import MatrixSpace
 from sage.rings.integer_ring import ZZ
+from sage.libs.flint.fmpz_poly_sage cimport Fmpz_poly
+
+
+cpdef Fmpz_poly chain_poly(list positions):
+    r"""
+    Compute the chain polynomial of a poset.
```
Comment (dcoudert) [inline src/sage/combinat/posets/hasse_cython_flint.pyx]: Return the chain polynomial of a poset. ?

**Review by dcoudert — CHANGES_REQUESTED** [review]
otherwise LGTM.

File: src/sage/combinat/posets/hasse_cython_flint.pyx (hunk 1 above)
Comment (fchapoton) [reply src/sage/combinat/posets/hasse_cython_flint.pyx]: voila

**Review by dcoudert — APPROVED** [review]
LGTM.

**Pull request merged**